### ISSUE TITLE
Add extensions option for better DX experience

### ISSUE THREAD

**Issue by alexander-akait** [stylelint/stylelint#3411]
> What is the problem you're trying to solve?

Linting all supported syntax (include non css - `html`, `markdown` and etc).
Now i use:

```shell
stylelint '{**/*,*}.{css,sass,scss,md,html}'
```

But it is very not convenient:
1. Long line in `package.json` with all supported extensions.
2. When we add new syntax and it is require add new extension to command
3. Some syntaxes support multiple extensions, example `markdown` support md, markdown, mdown, mdwn, mkd, mkdn, mkdown, ronn, workbook extensions. HTML and CSS-in-JS also have many extensions (future release). CSS

> What solution would you like to see?

It can be solved using:
1. Contain supported `extensions` inside package.
2. Use `stylelint .` (if extensions doesn't exists in glob patter(s) we filter all files by `extensions` array)
3. Add flag `--ext` (or `--extensions`) to CLI to allow specify extensions (for example `eslint` have this already).
4. Allow set `extensions` array using `stylelint.config.js`

`prettier` will do same in `2` version.

/cc @stylelint/contributors

**Comment by filipekiss** [comment]
I guess a default set of extensions combined with `stylelint` with no arguments is a great method. I've used `xo` in a project and running `xo` will lint all files supported (`.js`, `.jsx`) for the defaults. And then the user may pass the extensions either via glob, as it is today, using an `--extensions` option and also setting the `extensions` array in `stylelint.config.js`.

I also think that the `--extensions` options should merge with the default extensions by default and we should have an option (and a setting) like `--no-default-extensions` or something to disable the extensions.

**Comment by jeddy3** [comment]
SGTM. 

Being able to use `stylelint .` will allow users to align their CLI behaviour with that of the Visual Studio Code extension.

**Comment by Mouvedia** [comment]
I started working on this. Ill send a PR draft shortly.

**Comment by ybiquitous** [comment]
I have extracted #6689 from this issue. Please see the discussion on #6617 for details.

**Comment by ybiquitous** [comment]
For now, I don't think a new config property for file extensions is necessary. Because:

- the `--ext` flag ([#6689](https://github.com/stylelint/stylelint/issues/6689)) covers most use-cases
- a new property may conflict with a property suggested on [#3860](https://github.com/stylelint/stylelint/issues/3860)

**Comment by Mouvedia** [comment]
related: [ESLint RFC decision to remove `--ext`](https://github.com/eslint/rfcs/blob/main/designs/2019-config-simplification/README.md?plain=1#L550)

**Comment by Mouvedia** [comment]
> 1. Contain supported `extensions` inside package.
> 2. Use `stylelint .` (if extensions doesn't exists in glob patter(s) we filter all files by `extensions` array)
> 3. Add flag `--ext` (or `--extensions`) to CLI to allow specify extensions (for example `eslint` have this already).
> 4. Allow set `extensions` array using `stylelint.config.js`

@alexander-akait @stephenliu1944
2, 3 and 4 are covered by #6617
1 is now the one covered by this issue

There's a discussion going on on #6689. Please engage in it if you don't want 2, 3 and 4 to be dropped.
To summarize the move to a flat config, inspired by ESLint, is the main argument against it.

**Comment by jeddy3** [comment]
Closing in favour of https://github.com/stylelint/stylelint/issues/3860
